docs/getting-started/quickstart-programming-python.md

```diff
@@ -0,0 +1,325 @@
+---
+sidebar_position: 2
+title: DBOS Python Guide
+---
+
+This tutorial assumes you have finished the [Quickstart](./quickstart.md) and you have an app running locally.
+In this guide we'll modify that app to reliably record events across two different systems: Postgres and a third-party API.
+This app will write to both systems consistently, even if interrupted or restarted at any point.
+
+## 1. Serving HTTP Requests
+
+Let's first use [FastAPI](https://fastapi.tiangolo.com/) to implement a simple HTTP endpoint to greet friends. In your app folder, change the file `main.py` to contain only the following:
+
+```python
+from dbos import DBOS
+from fastapi import FastAPI
+from fastapi.responses import HTMLResponse
+
+app = FastAPI()
+DBOS(app)
+
+
+@app.get("/")
+def readme() -> HTMLResponse:
+    readme = """<html><body><p>
+      Welcome! Visit the route /greeting/:name to be greeted!<br>
+      For example, visit <a href="/greeting/dbos">/greeting/dbos</a>.<br>
+      </p></body></html>
+      """
+    return HTMLResponse(readme)
+
+@app.get("/greeting/{name}")
+def greet(name: str) -> str:
+    return f"Thank you for being awesome, {name}!"
+```
+
+Start your application with `dbos start`. You should see an output similar to:
+
+```shell
+INFO:     Started server process [191280]
+INFO:     Waiting for application startup.
+14:21:20 [    INFO] (dbos:admin_sever.py:31) Starting DBOS admin server on port 3001
+14:21:20 [    INFO] (dbos:dbos.py:317) DBOS initialized
+INFO:     Application startup complete.
+14:21:20 [    INFO] (dbos:system_database.py:657) Listening to notifications
+INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
+```
+
+To see that your application is working, visit this URL in your browser: [http://localhost:8000/greeting/Mike](http://localhost:8000/greeting/Mike). You should see the message `Thank you for being awesome, Mike!`. If you replace Mike with a different name, your application will greet that name instead. 
+
+## 2. Creating Database Tables
+
+Let's make a database table to record greetings.
+We strongly recommend managing database tables using [schema migrations](https://en.wikipedia.org/wiki/Schema_migration).
+DBOS supports any Python database migration tool, but in this tutorial we'll use [Alembic](https://alembic.sqlalchemy.org/en/latest/).
+First, let's define our new table.
+Change the file `schema.py` to contain only the following:
+
+```python
+from sqlalchemy import Column, Integer, MetaData, Table, Text
+
+metadata = MetaData()
+
+greetings = Table(
+    "greetings",
+    metadata,
+    Column("id", Integer, primary_key=True, autoincrement=True),
+    Column("name", Text),
+    Column("note", Text),
+)
+```
+
+Then, let's [autogenerate](https://alembic.sqlalchemy.org/en/latest/autogenerate.html) a migration from our schema file.
+Run the following commands:
+
+```
+dbos migrate
+alembic revision --autogenerate -m "greeting_guestbook"
+```
```

Comment:
This is trickier than represented here, and at least deserves a note that you have to have your DB up and have run the migration to this point for it to work correctly.

Reply:
Assuming the database is up (which it has to be for the previous commands to have worked) and they haven't been manually editing it, the first command will get it in the right state for the second command to work properly. 

It might be safer, if more complex, to not recommend autogenerate and instead tell them to make a blank migration and tell them what to copy-paste into it.

Reply:
We can put a "learn more" link to alembic at the end.